src/herbivoresAndCarnivores.js

```diff
@@ -1,16 +1,42 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const ANIMAL_DIED = 0;
+const BITE_DEMAGE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = FULL_HEALTH;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && animal instanceof Herbivore) {
+      animal.health -= BITE_DEMAGE;
+    }
+
+    if (animal.health === ANIMAL_DIED) {
```

Comment:
If `animal.health` is less than 0, the condition will not be fulfilled, but it should be.